### PR TITLE
CA-403101: Keep host.last_update_hash for host joined a pool

### PR DESCRIPTION
**Pull request by gangj**

### CODE REVIEW

**Comment by robhoes** [comment]
Commit message: the field is `host.last_update_hash`.

File: ocaml/idl/datamodel_host.ml [hunk 1]
```diff
@@ -1291,6 +1291,15 @@ let create_params =
     ; param_release= dundee_release
     ; param_default= Some (VDateTime Date.epoch)
     }
+  ; {
+      param_type= String
+    ; param_name= "last_update_hash"
+    ; param_doc=
+        "The SHA256 checksum of updateinfo of the most recently applied update \
+         on the host"
+    ; param_release= numbered_release "2.21.0-next"
```
Comment (robhoes) [inline ocaml/idl/datamodel_host.ml:1300]: We are at version 24.39.0 already.

**Review by robhoes — APPROVED** [review]

File: ocaml/idl/datamodel_host.ml (hunk 1 above)
Comment (gangj) [reply ocaml/idl/datamodel_host.ml:1300]: I see in "numbered_release", the parameter "version" is passed to “get_product_releases”: https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel_common.ml#L379
In "get_product_releases", it goes through "release_order": https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel_types.ml#L360, which is the XenServer releases.
So I think the string here should be "XenServerVersion-next", not the "XapiVersion-next"?

Comment (robhoes) [reply ocaml/idl/datamodel_host.ml:1300]: We don't use XS version here anymore, just xapi versions. It's the same as for the lifecycle fields, but the type is a little different.

Comment (gangj) [reply ocaml/idl/datamodel_host.ml]: OK, I see.

**Review by minglumlu — APPROVED** [review]

**Pull request merged**